<script>{
		"title":
			"deferred.fail()",
		"excerpt":
			" Add handlers to be called when the Deferred object is rejected. ",
		"termSlugs": {
			"category": [
				"deferred-object","version/1.5"
			]
		}
	}</script><h2 class="section-title">
<span class="name">deferred.fail( failCallbacks [, failCallbacks ] )</span><span class="returns">Returns: Deferred</span>
</h2><div class="entry-wrapper">
<p class="desc"><strong>Description: </strong> Add handlers to be called when the Deferred object is rejected. </p>
<ul class="signatures"><li class="signature">
<h4 class="name">
<span class="version-details">version added: 1.5</span>deferred.fail( failCallbacks [, failCallbacks ] )</h4>
<ul>
<li>
<div><strong>failCallbacks</strong></div>
<div>Type: Function()</div>
<div>
        A function, or array of functions, that are called when the Deferred is rejected.
      </div>
</li>
<li>
<div><strong>failCallbacks</strong></div>
<div>Type: Function()</div>
<div>
        Optional additional functions, or arrays of functions, that are called when the Deferred is rejected.
      </div>
</li>
</ul>
</li></ul>
<div class="longdesc" id="entry-longdesc">
    <p>The <code>deferred.fail()</code> method accepts one or more arguments, all of which can be either a single function or an array of functions. When the Deferred is rejected, the failCallbacks are called. Callbacks are executed in the order they were added. Since <code>deferred.fail()</code> returns the deferred object, other methods of the deferred object can be chained to this one, including additional <code>deferred.fail()</code> methods. The failCallbacks are executed using the arguments provided to the <code>deferred.reject()</code> or <code>deferred.rejectWith()</code> method call in the order they were added. For more information, see the documentation for Deferred object.</p>
  </div>
<section class="entry-examples" id="entry-examples"><header><h2 class="underline">Example:</h2></header><div class="entry-example" id="example-0">
<h4><span class="desc">Since the <code>jQuery.get</code> method returns a jqXHR object, which is derived from a Deferred, you can attach a success and failure callback using the <code>deferred.done()</code> and <code>deferred.fail()</code> methods.</span></h4>
<div class="syntaxhighlighter javascript ">
	<table>
		<tbody>
			<tr>
				<td class="gutter">
					
						<div class="line n1">1</div>
					
						<div class="line n2">2</div>
					
						<div class="line n3">3</div>
					
						<div class="line n4">4</div>
					
						<div class="line n5">5</div>
					
						<div class="line n6">6</div>
					
						<div class="line n7">7</div>
					
				</td>
				<td class="code">
					<pre><div class="container"><div class="line"><code>$.get( <span class="string">"test.php"</span> )</code></div></div><div class="container"><div class="line"><code>  .done(<span class="keyword">function</span>() {</code></div></div><div class="container"><div class="line"><code>    alert( <span class="string">"$.get succeeded"</span> );</code></div></div><div class="container"><div class="line"><code>  })</code></div></div><div class="container"><div class="line"><code>  .fail(<span class="keyword">function</span>() {</code></div></div><div class="container"><div class="line"><code>    alert( <span class="string">"$.get failed!"</span> );</code></div></div><div class="container"><div class="line"><code>  });</code></div></div></pre>
				</td>
			</tr>
		</tbody>
	</table>
</div>

</div></section>
</div>